src/app/settings/secrets/AddSecretDialog.tsx

```diff
@@ -0,0 +1,231 @@
+	Input,
+	useToast
+} from "@zenml-io/react-component-library";
+import PlusIcon from "@/assets/icons/secret-add.svg?react";
+import DeleteIcon from "@/assets/icons/secret-delete.svg?react";
+import EyeIcon from "@/assets/icons/eye.svg?react";
+import EyeOffIcon from "@/assets/icons/eye-off.svg?react";
+import { useQueryClient } from "@tanstack/react-query";
+import { useCreateSecretMutation } from "@/data/secrets/create-secret-query";
+import { isFetchError } from "@/lib/fetch-error";
+
+interface KeyValue {
+	key: string;
+	value: string;
+}
+
+interface Workspace {
+	id: string;
+}
+
+export function AddSecretDialog({ id, workspace }: { id: string; workspace: Workspace }) {
+	const [open, setOpen] = useState(false);
+
+	return (
+		<Dialog
+			open={open}
+			onOpenChange={(val) => {
+				setOpen(val);
+			}}
+		>
+			<DialogTrigger asChild>
+				<Button className="shrink-0" intent="primary">
+					Add secret
+				</Button>
+			</DialogTrigger>
+			<DialogContent className="mx-auto w-[90vw] max-w-[744px]">
+				<DialogHeader>
+					<DialogTitle>Register New Secret</DialogTitle>
+				</DialogHeader>
+				<AddSecret userId={id} setOpen={setOpen} workspaceId={workspace.id} />
+			</DialogContent>
+		</Dialog>
+	);
+}
+
+export function AddSecret({
+	userId,
+	setOpen,
+	workspaceId
+}: {
+	userId: string;
+	setOpen: (open: boolean) => void;
+	workspaceId: string;
+}) {
+	const [secretName, setSecretName] = useState("");
+	const [keysValues, setKeysValues] = useState<KeyValue[]>([{ key: "", value: "" }]);
+	const [visibleIndex, setVisibleIndex] = useState<number | null>(null);
+
+	const addKeyValuePair = () => {
+		const allFilled = keysValues.every((pair) => pair.key !== "" && pair.value !== "");
+		if (allFilled) {
+			setKeysValues([...keysValues, { key: "", value: "" }]);
+		} else {
+			alert("Please fill out all previous key-value pairs before adding a new one.");
+		}
+	};
+
+	useEffect(() => {});
+
+	const { toast } = useToast();
+	const queryClient = useQueryClient();
+	const { mutate } = useCreateSecretMutation({
+		onError(error) {
+			if (isFetchError(error)) {
+				toast({
+					status: "error",
+					emphasis: "subtle",
+					description: error.message,
+					rounded: true
+				});
+			}
+		},
+		onSuccess() {
+			queryClient.invalidateQueries({ queryKey: ["secrets"] });
+			setOpen(false);
+			setSecretName("");
+			setKeysValues([{ key: "", value: "" }]);
+		}
+	});
+
+	const handleInputChange = (index: number, event: React.ChangeEvent<HTMLInputElement>) => {
+		const { name, value } = event.target;
+		const updatedKeysValues = keysValues.map((pair, idx) =>
+			idx === index ? { ...pair, [name]: value } : pair
+		);
+		setKeysValues(updatedKeysValues);
+	};
+
+	const removeKeyValuePair = (index: number) => {
+		if (keysValues.length > 1) {
+			const updatedKeysValues = [...keysValues];
+			updatedKeysValues.splice(index, 1);
+			setKeysValues(updatedKeysValues);
+		}
+	};
+
+	const togglePasswordVisibility = (index: number) => {
```

Comment:
Can we please use `react-hook-form` for form hanlding with a zodschema, so it's consistent across the application :+1:. With the `useFieldArray` from it, the entire form handling should get a lot easier, and less prone to errors, because we dont need to handle so much state manually

Reply:
Done